### PR TITLE
Respa admin: fix period deletion functionality

### PR DESCRIPTION
**Pull request by danjacob-anders**
https://andersinno.atlassian.net/jira/software/c/projects/TTVA/boards/11?modal=detail&selectedIssue=TTVA-137

### CODE REVIEW

File: respa_admin/templates/respa_admin/common/_periods_accordion.html [hunk 1]
```diff
@@ -42,6 +42,10 @@
          role="tabpanel"
          id="collapse{{ id }}"
          aria-labelledby="heading{{ id }}">
+        <input type="hidden"
```
Comment (danjacob-anders) [inline respa_admin/templates/respa_admin/common/_periods_accordion.html:45]: Later versions of Django (4.x?) allow you to specify `deletion_widget` for a `Formset` class, so we could just do `{{ form.DELETE }}`.

File: respa_admin/forms.py [hunk 2]
```diff
@@ -374,14 +374,15 @@ def save(self, commit=True):
...
             for form in self.forms:
-                period = form.save(commit=commit)
-                if period.template_src and (
-                    form.has_changed() or form.days.has_changed()
-                ):
-                    period.template_src = None
-                    period.save()
-                if hasattr(form, "days"):
-                    form.days.save(commit=commit)
+                if form not in self.deleted_forms:
```
Comment (danjacob-anders) [inline respa_admin/forms.py:377]: `Formset.deleted_forms` tracks the form instances flagged for deletion. We need to skip these here or the deleted Periods will be added back in again as new instances.

File: respa_admin/static_src/js/periods.js [hunk 3]
```diff
@@ -517,20 +517,9 @@ function removePeriodEventHandlers(periodItem) {
 }
 
 function removePeriod(periodItem) {
+  // checks the DELETE input flag for this item.
   if (periodItem) {
-    periodItem.remove();
```
Comment (danjacob-anders) [inline respa_admin/static_src/js/periods.js:521]: We want to hide the Period form and just flag it for deletion by setting the `DELETE` hidden input to a non-empty value. Django will then process the form and handle the deletion normally, and the original order of existing forms is maintained.

**Review by saaste — APPROVED** [review]
By looking at your PR I relized how little I still understand about Django 😅 

Anyways, the code looks good to me. I'm assuming JS is expected to be enabled so relying on it is fine in this use-case ☺️

**Pull request merged**